core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,591 @@
+    # to None since initially, new reports received will not be assigned to a
+    # ticket.
+    ticket_id = datastore_services.StringProperty(default=None, indexed=True)
+    # Datetime in UTC of when the report was submitted by the user on their
+    # device. This may be much earlier than the model entity's creation date if
+    # the report was locally cached for a long time on an Android device.
+    submitted_on = datastore_services.DateTimeProperty(
+        required=True, indexed=True)
+    # The type of feedback for this report; this can be an arbitrary string
+    # since iterations of the report structure may introduce new types and we
+    # cannot rely on the backend updates to fully sync with the frontend report
+    # updates.
+    report_type = datastore_services.StringProperty(required=True, indexed=True)
+    # The category that this feedback is for.
+    category = datastore_services.StringProperty(required=True, indexed=True)
+    # The version of the app; on Android this is the package version name (e.g.
+    # 1.0-release-arm...) and on web this is the release version (e.g. 3.0.8).
+    platform_version = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The user's country locale represented as a ISO-3166 code; the locale is
+    # determined by the user's Android device settings.
+    device_country_locale_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The entry point location that the user is accessing the feedback report
+    # from on both web & Android devices. On Android, this could be
+    # navigation_drawer, lesson_player, revision_card, or crash.
+    entry_point = datastore_services.StringProperty(required=True, indexed=True)
+    # Additional topic / story / exploration IDs that may be collected depending
+    # on  the entry_point used to send the report; lesson player will have
+    # topic_id, story_id, and exploration_id, while revision card while have
+    # topic_id and subtopic_id.
+    entry_point_topic_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_story_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_exploration_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_subtopic_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+
+    # The text language on Oppia set by the user in its ISO-639 language code;
+    # this is set by the user in Oppia's app preferences on all platforms.
+    text_language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The audio language ISO-639 code on Oppia set by the user; this is set in
+    # Oppia's app preferences on all platforms.
+    audio_language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+
```

Comment:
Nit here & elsewhere: there are some inconsistencies in spacing between properties. I suggest keeping the spacing consistent with other models.

Reply:
Ack, done, removing extra spaces, though I didn't see much of a convention for why some models had empty lines between the listed fields.